ansible_ai_connect/ai/api/model_pipelines/langchain/pipelines.py

```diff
@@ -63,6 +63,46 @@
 HUMAN_MESSAGE_TEMPLATE = "{prompt}"
 
 
+def unwrap_role_answer(message: str | BaseMessage, create_outline: bool) -> tuple[str, list, str]:
+    task: str = ""
+    if isinstance(message, BaseMessage):
+        if (
+            isinstance(message.content, list)
+            and len(message.content)
+            and isinstance(message.content[0], str)
+        ):
+            task = message.content[0]
+        elif isinstance(message.content, str):
+            task = message.content
+    elif isinstance(message, str):
+        # Ollama currently answers with just a string
+        task = message
+    if not task:
+        raise ValueError
+
+    parts = task.split('!-!-!')
+
+    role = parts[0]
```

Comment:
You check (later) that `len(parts)>3` but you don't check that `parts` is sufficient to access indexes `0` ,`1` or `2`. Would it be prudent to have a check here too?

Reply:
It is because outline (part 3) is optional and part one and two always expected to be in the answer. But good point, let's return empty value instead of error if model return something unexpected.